### PR TITLE
Feat/rtenu 125 file management

### PR DESCRIPTION
**Pull request by saneDG**
- Add form for uploading file
- Add few new components for common use
- Add translations
- TODO: handle file upload edge cases and tests

### CODE REVIEW

File: packages/frontend/src/services/FileUploadService.ts [hunk 1]
```diff
@@ -1,15 +1,25 @@
...
+    form.append('name', name);
+    form.append('filedata', file);
+    form.append('nodeType', 'cm:content');
+    const options = {
+      method: 'POST',
+      body: form,
+    };
+    await fetch(`/api/alfresco/file/${parentNode}`, options)
+      .then((response) => response.json())
+      .then((data) => console.log(data))
```
Comment (mehtis) [inline packages/frontend/src/services/FileUploadService.ts:20]: Console.log?

File: packages/frontend/src/components/FileInput/styles.css [hunk 2]
```diff
@@ -0,0 +1,9 @@
+.file-input {
```
Comment (mehtis) [inline packages/frontend/src/components/FileInput/styles.css:1]: Khanh knows more about this stuff, but why is this as a css file while some other stuff is inline and some is using styled? Not sure if this is on purpose or not, just something that caught my eye.

File: packages/frontend/src/components/Files/FileUploadDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+  categoryName: string;
+}
+
+export const FileUploadDialog = ({ categoryName }: FileUploadProps) => {
+  const { t } = useTranslation(['common']);
+
+  const [file, setFile] = useState<File>();
+  const [name, setName] = useState<string>('');
+  const [description, setDescription] = useState<string>('');
+  const [dialogPhase, setPhase] = useState<string>('select-file');
```
Comment (mehtis) [inline packages/frontend/src/components/Files/FileUploadDialog.tsx:31]: This should be enum or something in the future. Makes using switch-case easier.

File: packages/frontend/src/components/Files/FileUploadDialog.tsx [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+                    <TextField
+                      sx={{ margin: '4px 0 0 0' }}
+                      className="form-text-field"
+                      onChange={(e) => setDescription(e.target.value)}
+                      fullWidth
+                    ></TextField>
+                  </Collapse>
+                </div>
+              ) : (
+                'File not selected '
```
Comment (mehtis) [inline packages/frontend/src/components/Files/FileUploadDialog.tsx:128]: Translations

File: packages/frontend/src/pages/Others/ManagementReports/index.tsx [hunk 5]
```diff
@@ -1,14 +1,20 @@
...
 export const ManagementReports = () => {
   const categoryNames = getSubCategoryData();
+  let [dialogOpen, setDialogOpen] = useState(false);
 
   return (
     <ProtectedContainerWrapper>
       <PageTitleWrapper>{categoryNames.MANAGEMENT_REPORTS}</PageTitleWrapper>
       <CategoryFiles categoryName={categoryNames.MANAGEMENT_REPORTS} />
+      {/* button to test file upload */}
+      <button onClick={() => setDialogOpen(true)}>LATAA TIEDOSTO</button>
```
Comment (mehtis) [inline packages/frontend/src/pages/Others/ManagementReports/index.tsx:16]: Translations

File: packages/frontend/src/pages/Others/ManagementReports/index.tsx [hunk 6]
```diff
@@ -1,14 +1,20 @@
...
+      {dialogOpen ? <FileUploadDialog categoryName={categoryNames.MANAGEMENT_REPORTS}></FileUploadDialog> : <></>}
```
Comment (mehtis) [inline packages/frontend/src/pages/Others/ManagementReports/index.tsx:17]: Can the second return option just be null?

**Review by mehtis — APPROVED** [review]
I'm approving this so that you can test this, but I'm expecting you to respond to all my comments one way or another (i.e. code changes or replies).

File: packages/frontend/src/services/FileUploadService.ts (hunk 1 above)
Comment (NinaDang97) [reply packages/frontend/src/services/FileUploadService.ts:20]: I'd like to have notification here (e.g. toaster pops up) in either success or fail case

File: packages/frontend/src/services/FileUploadService.ts [hunk 7]
```diff
@@ -1,15 +1,25 @@
...
+  if (file) {
+    const form = new FormData();
+    form.append('name', name);
+    form.append('filedata', file);
+    form.append('nodeType', 'cm:content');
+    const options = {
+      method: 'POST',
+      body: form,
+    };
+    await fetch(`/api/alfresco/file/${parentNode}`, options)
```
Comment (NinaDang97) [inline packages/frontend/src/services/FileUploadService.ts:18]: Again about consistency throughout the project, we already have axios package installed, and I think replace fetch with axios will remove 1 line of response.json()

File: packages/frontend/src/components/FileInput/styles.css (hunk 2 above)
Comment (saneDG) [reply packages/frontend/src/components/FileInput/styles.css]: Solved this by using MUI TextField component

File: packages/frontend/src/pages/Others/ManagementReports/index.tsx (hunk 5 above)
Comment (saneDG) [reply packages/frontend/src/pages/Others/ManagementReports/index.tsx:16]: Will leave this as this is since it will be removed

**Review by NinaDang97 — APPROVED** [review]
